### PR TITLE
fix: pasting blockquote

### PR DESCRIPTION
**Pull request by IBratWurst**
Break a paragraph when pasting a blockquote in between

### CODE REVIEW

**Comment by cezarsmpio** [comment]
Is it possible to add tests?

**Comment by IBratWurst** [comment]
> Is it possible to add tests?

Yes, I'm adding some cases

File: packages/rich-text/src/plugins/Quote/createQuotePlugin.ts [hunk 1]
```diff
@@ -39,5 +41,46 @@ export function createQuotePlugin(): RichTextPlugin {
...
+
+      editor.insertFragment = (fragment) => {
+        const startingNode = fragment.length && fragment[0];
+        const startsWithBlockquote =
+          Element.isElement(startingNode) && startingNode.type === BLOCKS.QUOTE;
+
+        let cursorEntry = getAbove(editor);
+        const cursorIsAtParagraph =
+          cursorEntry &&
+          TEXT_CONTAINERS.includes(cursorEntry?.[0]?.type) &&
```
Comment (z0al) [inline packages/rich-text/src/plugins/Quote/createQuotePlugin.ts:55]: Are you trying to see if we are inside a text container (paragraph or heading)? If yes, then you may wan to try this:

```javascript
const containerEntry = getAbove(editor, {
  match: {
    type: TEXT_CONTAINERS,
  }
})
```

**Review by z0al — COMMENTED** [review]
I tested the changes [here](https://deploy-preview-1035--contentful-field-editors.netlify.app/rich-text). I noticed two things:

* We sometimes paste the quote twice (when some text is selected)
* Undo & redo are behaving strangely. It feels like when undo-ing we skip some steps

File: packages/rich-text/src/plugins/Quote/createQuotePlugin.ts (hunk 1 above)
Comment (IBratWurst) [reply packages/rich-text/src/plugins/Quote/createQuotePlugin.ts]: Updated 👍🏼

**Comment by IBratWurst** [comment]
> I tested the changes [here](https://deploy-preview-1035--contentful-field-editors.netlify.app/rich-text). I noticed two things:
> 
> * We sometimes paste the quote twice (when some text is selected)
> * Undo & redo are behaving strangely. It feels like when undo-ing we skip some steps

@z0al do you have some examples steps to reproduce it?

File: packages/rich-text/src/plugins/Quote/createQuotePlugin.ts (hunk 1 above)
Comment (z0al) [reply packages/rich-text/src/plugins/Quote/createQuotePlugin.ts]: Cool. Do you want to apply the same changes to line 52?

**Comment by z0al** [comment]
> @z0al do you have some examples steps to reproduce it?

I tested this again. Couldn't reproduce the first issue. The second issue is reproducible but I also noticed it wasn't really caused by this PR

https://user-images.githubusercontent.com/12673605/155117028-b508eb08-afd9-4dfe-a670-e6580d5fa385.mp4

File: packages/rich-text/src/plugins/Quote/createQuotePlugin.ts [hunk 2]
```diff
@@ -39,5 +41,48 @@ export function createQuotePlugin(): RichTextPlugin {
...
+
+      editor.insertFragment = (fragment) => {
+        const startingNode = fragment.length && fragment[0];
+        const startsWithBlockquote =
+          Element.isElement(startingNode) && startingNode.type === BLOCKS.QUOTE;
+
+        const cursorEntry = getAbove(editor);
+        const cursorIsAtParagraph =
+          cursorEntry &&
+          TEXT_CONTAINERS.includes(cursorEntry?.[0]?.type) &&
```
Comment (z0al) [inline packages/rich-text/src/plugins/Quote/createQuotePlugin.ts:55]: You probably want to do a similar `getAbove(editor, {match: ...})` here

**Review by z0al — APPROVED** [review]
It works based on my testing 👍🏻

**Pull request merged**